packages/@react-spectrum/combobox/test/ComboBox.test.js

```diff
@@ -0,0 +1,1486 @@
+        testComboBoxOpen(combobox, button, listbox, 0);
+      });
+
+      it('opens for touch', () => {
+        let {getByRole} = renderComboBox({});
+
+        let button = getByRole('button');
+        let combobox = getByRole('combobox');
+        expect(document.activeElement).not.toBe(combobox);
+
+        act(() => {
+          fireEvent.touchStart(button, {targetTouches: [{identifier: 1}]});
+          fireEvent.touchEnd(button, {changedTouches: [{identifier: 1, clientX: 0, clientY: 0}]});
+          jest.runAllTimers();
+        });
+
+        expect(document.activeElement).toBe(combobox);
+        let listbox = getByRole('listbox');
+        expect(listbox).toBeTruthy;
+        expect(document.activeElement).toBe(combobox);
+      });
+    });
+
+    describe('keyboard input', function () {
+      it('opens the menu on down arrow press', function () {
+        let {getByRole} = renderComboBox();
+
+        let button = getByRole('button');
+        let combobox = getByRole('combobox');
+        combobox.focus();
+        expect(() => getByRole('listbox')).toThrow();
+
+        act(() => {
+          fireEvent.keyDown(combobox, {key: 'ArrowDown', code: 40, charCode: 40});
+          jest.runAllTimers();
+        });
+
+        let listbox = getByRole('listbox');
+        testComboBoxOpen(combobox, button, listbox, 0);
+      });
+
+      it('opens the menu on up arrow press', function () {
+        let {getByRole} = renderComboBox();
+
+        let button = getByRole('button');
+        let combobox = getByRole('combobox');
+        combobox.focus();
+        expect(() => getByRole('listbox')).toThrow();
+
+        act(() => {
+          fireEvent.keyDown(combobox, {key: 'ArrowUp', code: 38, charCode: 38});
+          jest.runAllTimers();
+        });
+
+        let listbox = getByRole('listbox');
+        testComboBoxOpen(combobox, button, listbox, 2);
+      });
+
+      it('opens the menu on user typing', function () {
+        let {getByRole} = renderComboBox();
+
+        let button = getByRole('button');
+        let combobox = getByRole('combobox');
+        combobox.focus();
+        expect(() => getByRole('listbox')).toThrow();
+
+        act(() => {
+          userEvent.type(combobox, 'Two');
+          jest.runAllTimers();
+        });
+
+        let listbox = getByRole('listbox');
+        expect(listbox).toBeVisible();
+        expect(button).toHaveAttribute('aria-expanded', 'true');
+        expect(button).toHaveAttribute('aria-controls', listbox.id);
+        expect(combobox).toHaveAttribute('aria-controls', listbox.id);
+
+        let items = within(listbox).getAllByRole('option');
+        expect(items).toHaveLength(1);
+        expect(items[0]).toHaveTextContent('Two');
+
+        expect(document.activeElement).toBe(combobox);
+        expect(combobox).toHaveAttribute('aria-activedescendant', items[0].id);
+      });
+
+      it('closes the menu if there are no matching items', function () {
```

Comment:
How about a test that opens menu on typing then closes it when no matches are found?

Reply:
This test actually does exactly that (it types in something matching then add a character that makes it so nothing matches anymore and thus the menu closes)